navbar in vue app

AfaqAhmed-XGrid opened the pull request:
## **Issue** #22   
     Title: Navbar in vue app

## Description
This pull request sets up the navbar in vue app:
=> Navbar: Used quasar and custom scss to design navbar that is fully responsive.
=> Http request: Used axios to make http request to get user data.
=> Routes Protection: Added routes protection
=> Scroll up button: Added scroll up button

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

AfaqAhmed-XGrid committed:
created navbar

created navbar, scroll up btn, callback function in routes that would redirect app to dashboard on accessing any unregistered route.

XGMqasim commented on ui_vue/src/App.vue:
```diff
@@ -44,14 +47,54 @@ watch(route, () => {
 </template>
+
+<style scoped lang="scss">
+@import './styles/breakpoints.scss';
```
Why the styles are not in a separate file?

AfaqAhmed-XGrid replied:
moved to the dedicated styling files

XGMqasim commented on ui_vue/src/components/TheNavBar.vue:
```diff
@@ -0,0 +1,212 @@
+  >
+    <q-scroll-area class="fit">
+      <q-list padding>
+        <q-item clickable v-ripple :active="route.path === '/dashboard'" to="/dashboard">
```
You can loop through all these elements

AfaqAhmed-XGrid replied:
Done

XGMqasim commented on ui_vue/src/composables/useAuthHttpRequests.ts:
```diff
@@ -135,5 +135,31 @@ export const useAuthHttpRequest = () => {
+    try {
+      const token = localStorage.getItem(constants.localStorage.userToken);
+      const response = await axios.get(baseURL + authEndPoint.getUser, {
+        headers: {
```
You can use axios interceptors for this otherwise you have to add headers in each request

AfaqAhmed-XGrid replied:
Done

XGMqasim reviewed (changes requested):
-1: Address all comments

AfaqAhmed-XGrid committed:
Added axios interceptor, moves css styling to separate file and used for loop to iterate over the navbar items

Created axios interceptor in main.ts file to add access token in all http request.

Moved css class to separate file rather than writing it in Navbar component file

Create the navbar items array in constants file and iterate over it in navbar component file

saif-xgrid reviewed (approved):
LGTM
